### PR TITLE
Add directory & configFile options to SwiftLint

### PR DESCRIPTION
**Pull request by sunshinejr**
Hey Ash!

So this is a workaround for #4. This shouldn't break any current configurations, as I've added default arguments. You can see a working copy of this one at [Harvey#11](https://github.com/Moya/Harvey/pull/11/files).

### CODE REVIEW

File: Tests/DangerSwiftLintTests/DangerSwiftLintTests.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@ class DangerSwiftLintTests: XCTestCase {
     override func setUp() {
         executor = FakeShellExecutor()
         // This is for me, testing. Uncomment if you're running tests locally.
-        FileManager.default.changeCurrentDirectoryPath("/Users/ash/bin/danger-swiftlint")
+        FileManager.default.changeCurrentDirectoryPath("/Users/lukaszmroz/Projects/OtherProjects/Libraries/danger-swiftlint")
```
Comment (freak4pc) [inline Tests/DangerSwiftLintTests/DangerSwiftLintTests.swift]: Probably need to revert this guy to the original ?

File: Tests/DangerSwiftLintTests/DangerSwiftLintTests.swift [hunk 2]
```diff
@@ -21,6 +21,30 @@ class DangerSwiftLintTests: XCTestCase {
         XCTAssertNotEqual(executor.invocations.dropFirst().count, 0)
     }
 
+    func testExecutesSwiftLintWithConfigWhenPassed() {
+        let configFile = "/Path/to/config/.swiftlint.yml"
```
Comment (freak4pc) [inline Tests/DangerSwiftLintTests/DangerSwiftLintTests.swift:24]: If this is a continuous test - shouldn't we just bundle a `.swiftlint.yml` mock in the repo ?

File: Sources/DangerSwiftLint/DangerSwiftLint.swift [hunk 3]
```diff
@@ -8,10 +8,10 @@ public struct SwiftLint {
     /// This is the main entry point for linting Swift in PRs using Danger-Swift.
     /// Call this function anywhere from within your Dangerfile.swift.
     @discardableResult
-    public static func lint() -> [Violation] {
+    public static func lint(directory: String? = nil, configFile: String? = nil) -> [Violation] {
```
Comment (freak4pc) [inline Sources/DangerSwiftLint/DangerSwiftLint.swift:11]: I'm not sure the `configFile` name is so pretty, I don't super-like it but I don't have a nicer idea at the moment 🤔

**Review by freak4pc — COMMENTED** [review]
Great change in general :) Helpful for external libraries for sure. 

Left two nits and one opinion which I'm not really sure what to do about 🤷‍♂️

File: Tests/DangerSwiftLintTests/DangerSwiftLintTests.swift (hunk 1 above)
Comment (sunshinejr) [reply Tests/DangerSwiftLintTests/DangerSwiftLintTests.swift]: Awww, I was pretty sure I didn't pushed this one, nice catch!

**Comment by freak4pc** [comment]
P.S no biggie @ashfurrow , but I think you forgot to turn on "Build forked pull requests", since CircleCI didn't actually do its thing here :)

File: Tests/DangerSwiftLintTests/DangerSwiftLintTests.swift (hunk 2 above)
Comment (sunshinejr) [reply Tests/DangerSwiftLintTests/DangerSwiftLintTests.swift:24]: Nah, this is just needed to check if the argument with this path was added.

File: Sources/DangerSwiftLint/DangerSwiftLint.swift (hunk 3 above)
Comment (sunshinejr) [reply Sources/DangerSwiftLint/DangerSwiftLint.swift:11]: I'm all ears for a better name 😄 didn't think of a better one as well

Comment (ashfurrow) [reply Sources/DangerSwiftLint/DangerSwiftLint.swift:11]: Not sure if a good alternative exists. It's the parameter used to specify the file where the config is 🙃

**Comment by ashfurrow** [comment]
Good catch, I have re-delivered the webhook event for the last commit in this PR to re-trigger the build.

**Review by ashfurrow — APPROVED** [review]
Looks great to merge, thanks! Just the CI failure at this point.

**Comment by sunshinejr** [comment]
Okay, fixed all! 

First there was a problem with initializing `JSONDecoder`, as dates in `danger-swift` now use `Date` type instead of a `String`. Then we needed to update our fixtures to cover a quite different `requested_reviewers` structure (dict with `users`/`teams` instead of an array now). 

It should be ready to go @ashfurrow

**Review by ashfurrow — APPROVED** [review]

**Comment by ashfurrow** [comment]
Cooooooool, tagged a 0.2.0 release with these changes. Thanks again!!

**Pull request merged**